### PR TITLE
 Merge kube-dns templates into a single file

### LINKED ISSUE #42832: Merge all of the kube-dns cluster yaml into a single file
We should merge all of the cluster/addons/dns/*.yaml.* into a single file (say `addon.yaml.base`)

This way all of the dependent scripts for cluster creation do not need to be modified when the addon is changed. This currently contributes to the maintenance cost of the addon as it is not immediately obvious all of the locations that need to be changed when an update to the addon is done.

@MrHohn

### PR DESCRIPTION
**Pull request by MrHohn**
**What this PR does / why we need it**: Merge all of the kube-dns cluster yamls into a single file.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #42832

**Special notes for your reviewer**:
/assign @bowei @shashidharatd 
cc @kevin-wangzefeng @euank @lhuard1A

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Review by shashidharatd — APPROVED** [review]
Looks good to me except a minor nit in log message. This PR was a long pending task and thanks to @MrHohn for completing it.
/lgtm

File: hack/local-up-cluster.sh [hunk 1]
```diff
@@ -755,18 +755,15 @@ EOF
...
-        # use kubectl to create kubedns deployment and service
-        ${KUBECTL} --kubeconfig="${CERT_DIR}/admin.kubeconfig" --namespace=kube-system create -f ${KUBE_ROOT}/cluster/addons/dns/kubedns-sa.yaml
-        ${KUBECTL} --kubeconfig="${CERT_DIR}/admin.kubeconfig" --namespace=kube-system create -f ${KUBE_ROOT}/cluster/addons/dns/kubedns-cm.yaml
-        ${KUBECTL} --kubeconfig="${CERT_DIR}/admin.kubeconfig" --namespace=kube-system create -f kubedns-deployment.yaml
-        ${KUBECTL} --kubeconfig="${CERT_DIR}/admin.kubeconfig" --namespace=kube-system create -f kubedns-svc.yaml
-        echo "Kube-dns deployment and service successfully deployed."
-        rm  kubedns-deployment.yaml kubedns-svc.yaml
+        # use kubectl to create kubedns addon
+        ${KUBECTL} --kubeconfig="${CERT_DIR}/admin.kubeconfig" --namespace=kube-system create -f kube-dns.yaml
+        echo "Kube-dns addone successfully deployed."
```
[the remark this answers is not in the document]
Comment (MrHohn) [reply hack/local-up-cluster.sh]: Thanks for taking the time. Fixed.

**Comment by MrHohn** [comment]
Thanks,
/assign @eparis 
for approval.

**Pull request merged**